### PR TITLE
Do not proceed loading ipfs resources if ipfs is disabled

### PR DESCRIPTION
**Pull request by cypt4**
Resolves https://github.com/brave/brave-browser/issues/23646

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/net/ipfs_redirect_network_delegate_helper.cc [hunk 1]
```diff
@@ -23,6 +23,10 @@ int OnBeforeURLRequest_IPFSRedirectWork(
     return net::OK;
```
Comment (iefremov) [inline browser/net/ipfs_redirect_network_delegate_helper.cc:23]: double check the incognito/guest/tor profiles

Comment (iefremov) [reply browser/net/ipfs_redirect_network_delegate_helper.cc:23]: also having a test would be nice

File: browser/net/ipfs_redirect_network_delegate_helper.cc [hunk 2]
```diff
@@ -19,10 +19,22 @@ namespace ipfs {
 int OnBeforeURLRequest_IPFSRedirectWork(
     const brave::ResponseCallback& next_callback,
     std::shared_ptr<brave::BraveRequestInfo> ctx) {
-  if (!ctx->browser_context || !brave::IsRegularProfile(ctx->browser_context))
+  bool hasIpfsScheme = IsIPFSScheme(ctx->request_url);
```
Comment (iefremov) [inline browser/net/ipfs_redirect_network_delegate_helper.cc:22]: const bool and `has_ipfs_scheme`

File: browser/net/ipfs_redirect_network_delegate_helper.cc [hunk 3]
```diff
@@ -19,10 +19,22 @@ namespace ipfs {
...
+  if (!ctx->browser_context) {
+    if (hasIpfsScheme) {
+      ctx->blocked_by = brave::kOtherBlocked;
```
Comment (iefremov) [inline browser/net/ipfs_redirect_network_delegate_helper.cc:25]: adding comments is much appreciated

File: browser/net/ipfs_redirect_network_delegate_helper.cc [hunk 4]
```diff
@@ -19,10 +19,22 @@ namespace ipfs {
...
+    }
     return net::OK;
+  }
+
   auto* prefs = user_prefs::UserPrefs::Get(ctx->browser_context);
-  if (IsIpfsResolveMethodDisabled(prefs)) {
+  bool ipfsDisabled = IsIpfsResolveMethodDisabled(prefs);
```
Comment (iefremov) [inline browser/net/ipfs_redirect_network_delegate_helper.cc:31]: code style

**Review by iefremov — APPROVED** [review]

**Comment by iefremov** [comment]
maybe @spylogsster wants to take a look

**Review by spylogsster — APPROVED** [review]
Nice work! LGTM ++

**Pull request merged**